opentracing-globaltracer/src/main/java/io/opentracing/GlobalTracer.java

```diff
@@ -0,0 +1,196 @@
+/**
+ * Copyright 2016-2017 The OpenTracing Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.opentracing;
+
+import io.opentracing.propagation.Format;
+
+import java.util.Iterator;
+import java.util.ServiceLoader;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * Forwards all methods to another tracer that can be configured in one of two ways:
+ * <ol>
+ * <li>Explicitly, calling {@link #register(Tracer)} with a configured tracer, or:</li>
+ * <li>Automatically using the Java {@link ServiceLoader} SPI mechanism to load a {@link Tracer} from the classpath.</li>
+ * </ol>
+ * <p>
+ * When the tracer is needed it is lazily looked up using the following rules:
+ * <ol type="a">
+ * <li>The last {@link #register(Tracer) registered} or {@link #update(UpdateFunction) updated} tracer
+ * always takes precedence.</li>
+ * <li>If no tracer was registered, one is looked up from the {@link ServiceLoader}.<br>
+ * The {@linkplain GlobalTracer} will not attempt to choose between implementations:</li>
+ * <li>If no single tracer service is found, the {@link io.opentracing.NoopTracer NoopTracer} will be used.</li>
+ * </ol>
+ */
+public final class GlobalTracer implements Tracer {
+    private static final Logger LOGGER = Logger.getLogger(GlobalTracer.class.getName());
```

Comment:
There was some discussion about aligning on SLF4j. This will be the first use of logging in this project, so we might want to make a decision on that.